pkg/instance/service.go

```diff
@@ -869,3 +812,106 @@ func (s service) destroy(instance *model.Instance) error {
 	}
 	return nil
 }
+
+type updateMessage struct {
+	EventType    watch.EventType `json:"eventType"`
+	DeploymentId uint            `json:"deploymentId"`
+	InstanceId   uint            `json:"instanceId"`
+	Status       Status          `json:"status"`
+}
+
+func (s service) ListenForClusterUpdates() {
+	fakeAdministrator := &model.User{
+		Groups: []model.Group{
+			{
+				Name: model.AdministratorGroupName,
+			},
+		},
+	}
+
+	groups, err := s.groupService.FindAll(fakeAdministrator, true)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, group := range groups {
+		g := group
+		go func() {
+			ks, err := NewKubernetesService(g.ClusterConfiguration)
+			if err != nil {
+				log.Printf("unable to get create new kubernetes service: %v", err)
+			}
+
+			for {
+				events, err := ks.watch(g.Name)
+				if err != nil {
+					log.Printf("error watching %q: %v", g.Name, err)
+				}
+
+				for e := range events {
+					s.handleEvent(e)
+				}
+			}
+		}()
+	}
+}
+
+func (s service) handleEvent(e watch.Event) {
+	pod, ok := e.Object.(*v1.Pod)
+	if !ok {
+		// TODO: Log entire e.Object... Entire e?
+		log.Println("unexpected type")
```

Comment:
return if you cannot handle the event. this is true for all code in this method.